extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/CaffeineCacheSupplier.java

```diff
@@ -0,0 +1,35 @@
+package io.quarkus.cache.runtime;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.cache.Cache;
+import io.quarkus.cache.runtime.caffeine.CaffeineCache;
+
+public class CaffeineCacheSupplier implements Supplier<Collection<CaffeineCache>> {
+
+    @Override
+    public List<CaffeineCache> get() {
+        CacheManagerImpl cacheManager = cacheManager();
```

Comment:
You don't need to inject the impl anymore:
```suggestion
        CacheManager cacheManager = cacheManager();
```

Reply:
OK fixed.